src/Middleware/WebSockets/src/WebSocketMiddleware.cs

```diff
@@ -146,7 +145,7 @@ public async Task<WebSocket> AcceptAsync(WebSocketAcceptContext acceptContext)
                     }
                 }
 
-                string key = string.Join(", ", _context.Request.Headers[HeaderNames.SecWebSocketKey]);
```

Comment:
This was odd, nothing in the RFC says this is valid.
Should we check that there is only one value?

Reply:
You're now using the implicit string converter that does the same thing here.

A multi-value header will fail the normal parse if there are multiple values anyways, don't worry about it.

Reply:
Is it allocating right now even for a single value?

Reply:
> Is it allocating right now even for a single value?

[No.](https://github.com/aspnet/Extensions/blob/b3a158ac055f1f8f76ed4a1a1c6ef860068cf63e/src/Primitives/src/StringValues.cs#L126)